### PR TITLE
feat: fetch data contract history endpoint

### PR DESCRIPTION
**Pull request by antouhou**
## Issue being fixed or feature implemented

Fetch contract history in SDK

## What was done?

- Added fetch method to dapi
- Added fetch method and related data structures to dapi-client, unit tests
- Added fetch method to SDK, unit tests
- Added check that the history is correct and fetchable in test suite

## How Has This Been Tested?

Run tests

## Breaking Changes

None

## Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added or updated relevant unit/integration/functional/e2e tests
- [x] I have made corresponding changes to the documentation

**For repository code-owners and collaborators only**
- [ ] I have assigned this pull request to a milestone

### CODE REVIEW

File: packages/dapi-grpc/protos/platform/v0/platform.proto [hunk 1]
```diff
@@ -269,20 +270,16 @@ message GetDataContractsResponse {
 
 message GetDataContractHistoryRequest {
   bytes id = 1;
-  optional uint32 limit = 2;
-  optional uint32 offset = 3;
-  optional uint64 start_at_seconds = 4;
+  uint32 limit = 2;
+  uint32 offset = 3;
+  uint64 start_at_seconds = 4;
```
Comment (QuantumExplorer) [inline packages/dapi-grpc/protos/platform/v0/platform.proto:275]: why not ms?

Comment (antouhou) [reply packages/dapi-grpc/protos/platform/v0/platform.proto:275]: Yeah, I've just checked, it is indeed in ms, not seconds. I guess I just misspelled it somewhere and then continued to name it the same afterwards

**Review by markin-io — APPROVED** [review]

File: packages/js-dash-sdk/src/SDK/Client/Platform/methods/contracts/history.ts [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+declare type ContractIdentifier = string | Identifier;
+
+/**
+ * Get contracts from the platform
+ *
+ * @param {ContractIdentifier} identifier - identifier of the contract to fetch
+ * @param startAtMs
+ * @param limit
+ * @param offset
```
Comment (shumkov) [inline packages/js-dash-sdk/src/SDK/Client/Platform/methods/contracts/history.ts:18]: Shall we define more high-level methods for SDK? Like get by block time and get by version and list all versions?

**Review by shumkov — CHANGES_REQUESTED** [review]

File: packages/js-dash-sdk/src/SDK/Client/Platform/methods/contracts/history.ts (hunk 2 above)
Comment (shumkov) [reply packages/js-dash-sdk/src/SDK/Client/Platform/methods/contracts/history.ts:18]: We made a separate task

**Review by shumkov — APPROVED** [review]
👍

**Pull request merged**